### PR TITLE
Fix #26846 (data-contract): handle non-Boolean semantics rule results and unsupported entity types

### LINKED ISSUE #26846: Semantic Rule validation fails when rule is empty – NullPointerException & UI does not persist semantic without modifying default rule
**Affected module**
Does it impact the UI, backend or Ingestion Framework? - Backend & UI

**Describe the bug**
- We are encountering an issue when creating a Semantic without modifying the default rule from the UI.
Problem 1 – Backend Error
When the Semantic Rule is left empty, the following error occurs:
```
Cannot invoke "java.lang.Integer.intValue()" because the return value of 
"org.openmetadata.schema.entity.datacontract.SemanticsValidation.getFailed()" is null
```
This suggests that SemanticsValidation.getFailed() is returning null, and the backend attempts to call intValue() without null handling.

Problem 2 – UI Behavior
From the UI perspective:
When creating a Semantic with only:
- Name
- Description
and saving without modifying the default rule, the Semantic is not persisted.

The semantics are only saved if:
The default rule is removed, OR a new rule is added, OR the default rule is modified
Ideally, when rules are untouched, the UI should send rules: null (or an empty list), allowing the Semantic to be persisted successfully.

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**
- Creating a Semantic with only name and description should persist successfully.
- If no rule is defined, backend should handle:
  - rules = null OR
  - rules = []
- Backend validation should not throw NullPointerException when validation fields are null.
- Default UI rule should not block persistence when not modified.

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by yan-3005**
- RuleEngine.validateRule: use instanceof pattern match instead of unsafe (Boolean) cast — prevents ClassCastException when JsonLogic returns a non-Boolean (e.g. empty-string rule evaluates to String "")
- DataContractRepository.validateSemantics: initialize failed/passed/total to 0 in catch block so SemanticsValidation never has null fields
- DataContractRepository.compileResult: null-check Integer before auto-unboxing getFailed() to prevent NPE in all three validation blocks
- DataContractRepository.isSupportedEntityType: add metric and apiService entity types; rename to public isEntityTypeSupported
- DataContractValidationApp: skip DP assets whose entity type is not supported instead of counting them as errors
- Tests: add RuleEngineTest unit tests for empty/non-Boolean/null rule results; add IT for empty semantics rule NPE regression

### Describe your changes:

Fixes #26846

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/DataContractRepository.java [hunk 1]
```diff
@@ -1201,7 +1203,12 @@ private SemanticsValidation validateSemantics(DataContract dataContract) {
                   .collect(Collectors.toList()));
 
     } catch (Exception e) {
-      LOG.error("Error during semantics validation", e);
+      LOG.error(
+          "Error during semantics validation for contract {}: {}",
+          dataContract.getFullyQualifiedName(),
+          e.getMessage(),
+          e);
+      validation.withFailed(0).withPassed(0).withTotal(0);
```
[the remark this answers is not in the document]
Comment (yan-3005) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/DataContractRepository.java]: Fixed
